### PR TITLE
Helm chart update

### LINKED ISSUE #684: Update PKO installer as necessary
- [x] fixups to be able to push the operator image
- [x] a new pulmui deploy app that uses the kustomization manifests in operator/config
- [x] updated helm chart
  - [x] cluster rbac
  - [x] namespaced rbac
  - [x] security context
  - [x] aggregated roles
- [x] updated install.yaml in deploy/yaml

### PR DESCRIPTION
**Pull request by EronWright**
### Proposed changes

Updates the Helm chart to install PKOv2, as similarly as possible to `operator/config/default`.

Details:
- adds an aggregation role (view/edit) for the Pulumi API groups
- tweaks the controller's resources such that limits equals resources to have "guaranteed" qos
- exposes the metrics port and the fileserver port
- supports two rbac modes for the controller - ClusterRole and Role

To install:
```
helm upgrade --install pulumi-kubernetes-operator ./deploy/helm/pulumi-operator
```

### Related issues (optional)

Closes #684

### CODE REVIEW

File: deploy/helm/pulumi-operator/Chart.yaml [hunk 1]
```diff
@@ -9,8 +9,8 @@ icon: https://www.pulumi.com/logos/brand/avatar-on-white.svg
 
 type: application
 
-version: 0.9.0
-appVersion: 1.14.0
+version: 2.0.0
+appVersion: "v2.0-devel"
```
Comment (EronWright) [inline deploy/helm/pulumi-operator/Chart.yaml:13]: Rationale for changing the version info:
-  `appVersion` is used as the image tag in deployment.yaml and should match our normal version string.  The prerelease target will replace the value with the tag.
- `version` is the chart version, and I would advocate for doing a major bump.

File: deploy/helm/pulumi-operator/values.yaml [hunk 2]
```diff
@@ -17,36 +17,28 @@ image:
...
-# -- The image pull secrets
-imagePullSecrets: ""
+# RBAC settings
+rbac:
+  # -- Specifies whether RBAC resources should be created
+  create: true
+  # -- Specifies whether cluster roles and bindings should be created
+  createClusterRole: true
+  # -- Specifies whether namespaced roles and bindings should be created
+  createRole: false
```
Comment (EronWright) [inline deploy/helm/pulumi-operator/values.yaml:37]: one either creates a clusterrole or a role for the controller manager, depending on whether you want a namepaced installation of the operator.  There's still some follow-up work needed, but the helm piece is in place.

File: deploy/helm/pulumi-operator/values.yaml [hunk 3]
```diff
@@ -71,41 +63,39 @@ podAnnotations: {}
...
   # resources, such as Minikube. If you do want to specify resources, uncomment the following
   # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
   limits:
-    cpu: 500m
-    memory: 5123Mi
+    cpu: 200m
+    memory: 128Mi
   requests:
-    cpu: 100m
+    cpu: 200m
```
Comment (EronWright) [inline deploy/helm/pulumi-operator/values.yaml:98]: Rationale is to ensure that the pod is assigned the ["Guaranteed"](https://kubernetes.io/docs/concepts/workloads/pods/pod-qos/#guaranteed) QoS class.

**Review by rquitales — APPROVED** [review]
LGTM
---
I assume users will also need to manually apply the new CRDs first before running: `helm upgrade --install pulumi-kubernetes-operator ./deploy/helm/pulumi-operator`?

File: deploy/helm/pulumi-operator/values.yaml [hunk 4]
```diff
@@ -143,11 +141,9 @@ extraSidecars: []
 serviceMonitor:
   # -- When set true then use a ServiceMonitor to configure scraping
   enabled: false
-  # Set the namespace the ServiceMonitor should be deployed
-  # namespace: monitoring
   # Set how frequently Prometheus should scrape
   # interval: 30s
-  # Set path to minecraft-exporter telemtery-path
+  # Set telemetry path
```
Comment (blampe) [inline deploy/helm/pulumi-operator/values.yaml:146]: :)

**Review by blampe — APPROVED** [review]
Awesome!

Obviously not something that we need to resolve now but it would be nice to more easily keep this in sync with the generated configs. There's https://github.com/kubernetes-sigs/kubebuilder/discussions/3074 which took me to https://github.com/arttor/helmify which looks interesting.

**Comment by rquitales** [comment]
> Awesome!
> 
> Obviously not something that we need to resolve now but it would be nice to more easily keep this in sync with the generated configs. There's [kubernetes-sigs/kubebuilder#3074](https://github.com/kubernetes-sigs/kubebuilder/discussions/3074) which took me to https://github.com/arttor/helmify which looks interesting.

@blampe the original helm chart for v1 was actually scaffolded using `helmify`.

**Pull request merged**